File: java/base/src/main/vsdk/toolkit/gui/feedback/parallel/ParallelProgressMonitorEvent.java
package vsdk.toolkit.gui.feedback.parallel;

public class ParallelProgressMonitorEvent {
    private final ParallelProgressMonitorCommand commandType;
    private final long numberOfElementsToProcess;

    public ParallelProgressMonitorEvent(
        ParallelProgressMonitorCommand commandType,
        long numberOfElementsToProcess)
    {
        this.commandType = commandType;
        this.numberOfElementsToProcess = numberOfElementsToProcess;
    }

    public ParallelProgressMonitorCommand getCommandType() {
        return commandType;
    }

    public long getNumberOfElementsToProcess() {
        return numberOfElementsToProcess;
    }
}

File: typescript/base/src/main/vsdk/toolkit/gui/feedback/parallel/ParallelProgressMonitorEvent.ts
import { ParallelProgressMonitorCommand } from "./ParallelProgressMonitorCommand.js";

export class ParallelProgressMonitorEvent {
    private readonly commandType: ParallelProgressMonitorCommand;
    private readonly numberOfElementsToProcess: bigint;

    public constructor(commandType: ParallelProgressMonitorCommand, numberOfElementsToProcess: bigint) {
        this.commandType = commandType;
        this.numberOfElementsToProcess = numberOfElementsToProcess;
    }

    public getCommandType(): ParallelProgressMonitorCommand {
        return this.commandType;
    }

    public getNumberOfElementsToProcess(): bigint {
        return this.numberOfElementsToProcess;
    }
}
